tests/integration/test_target_snowflake.py

```diff
@@ -874,6 +875,31 @@ def test_flush_streams_with_intermediate_flushes_on_all_streams(self, mock_emit_
         # Every table should be loaded correctly
         self.assert_logical_streams_are_in_snowflake(True)
 
+    @mock.patch('target_snowflake.emit_state')
+    @mock.patch('target_snowflake.datetime')
```

Comment:
Why mock `datetime` in an integration test?

The only reason `emit_state` is mocked is because it write to stdout and we need to capture that during the test.

Reply:
@Samira-El I'd appreciate any better ideas you have on how to test logic that's dependent on time

Reply:
I would suggest not to mock the time, instead have a minuscule value of `batch_wait_limit_seconds`, maybe fraction of a second, that should trigger a time based flush.

We won't know exactly how many flushes but we know for sure they would be > 1.

Reply:
I can do that if it's preferred, from prior experience it's just recipe for flakiness.

Reply:
I can also remove the whole thing if it doesn't fit well into your integration tests given these constraints, it's more or less a duplicate of the unit test.

Reply:
The integration tests use an actual Snowflake instance and the goal of it is mainly to test all the given streams & records have been persisted correctly. I think this time-based flush is more of an implmentation detail that can unit tests should cover and make sure behaves as expected.